packages/twenty-front/src/modules/favorites/hooks/useFavorites.ts

```diff
@@ -1,214 +1,159 @@
+              relationField.toRelationMetadata?.fromObjectMetadata
+                .nameSingular ?? '';
+
+            const objectRecordIdentifier =
+              getObjectRecordIdentifierByNameSingular(
+                relationObject,
+                relationObjectNameSingular,
+              );
+
+            return {
+              id: favorite.id,
+              recordId: objectRecordIdentifier.id,
+              position: favorite.position,
+              avatarType: objectRecordIdentifier.avatarType,
+              avatarUrl: objectRecordIdentifier.avatarUrl,
+              labelIdentifier: objectRecordIdentifier.name,
+              link: objectRecordIdentifier.linkToShowPage,
+            } as Favorite;
+          }
         }
-      },
-    [apolloClient, updateOneFavoriteMutation],
-  );
-
-  const deleteFavorite = useRecoilCallback(
-    ({ snapshot, set }) =>
-      async (favoriteIdToDelete: string) => {
-        const favoritesStateFromSnapshot = snapshot.getLoadable(favoritesState);
-        const favorites = favoritesStateFromSnapshot.getValue();
-        const idToDelete = favorites.find(
-          (favorite: Favorite) => favorite.recordId === favoriteIdToDelete,
-        )?.id;
-
-        await apolloClient.mutate({
-          mutation: deleteOneFavoriteMutation,
-          variables: {
-            idToDelete: idToDelete,
-          },
-        });
-
-        performOptimisticEvict('Favorite', 'id', idToDelete ?? '');
-
-        set(
-          favoritesState,
-          favorites.filter((favorite: Favorite) => favorite.id !== idToDelete),
-        );
-      },
-    [apolloClient, deleteOneFavoriteMutation, performOptimisticEvict],
-  );
 
-  const computeNewPosition = (destIndex: number, sourceIndex: number) => {
-    if (destIndex === 0) {
-      return favorites[destIndex].position / 2;
-    }
+        return favorite;
+      })
+      .toSorted((a, b) => a.position - b.position);
+  }, [
+    favorites,
+    favoriteRelationFields,
+    getObjectRecordIdentifierByNameSingular,
+  ]);
+
+  const createFavorite = (
+    targetObject: Record<string, any>,
+    targetObjectNameSingular: string,
+  ) => {
+    createOneFavorite({
+      [`${targetObjectNameSingular}Id`]: targetObject.id,
+      [`${targetObjectNameSingular}`]: targetObject,
+      position: favorites.length + 1,
+      workspaceMemberId: currentWorkspaceMember?.id,
+    });
+  };
 
-    if (destIndex === favorites.length - 1) {
-      return favorites[destIndex - 1].position + 1;
-    }
+  const deleteFavorite = (favoriteId: string) => {
+    deleteOneRecord(favoriteId);
+  };
 
-    if (sourceIndex < destIndex) {
+  const computeNewPosition = (destIndex: number, sourceIndex: number) => {
+    const moveToFirstPosition = destIndex === 0;
+    const moveToLastPosition = destIndex === favoritesSorted.length - 1;
+    const moveAfterSource = destIndex > sourceIndex;
+
+    if (moveToFirstPosition) {
+      return favoritesSorted[0].position / 2;
```

Comment:
Is it possible here to have `favoritesSorted[0] === 0` ? If so, wouldn't we have 2 favorites with the position 0 ?

Reply:
No because the first position will be `favorites.length + 1`, so the first favorite will always have position = 1.

We can end up with something like : `0.00000000005` after dividing by 2 a lot of time.

We could change the ordering system to something more complex but it was first designed to be implemented fast.

Here I just changed the predicates' naming, but we can create an issue to make this system more robust.